src/System.Diagnostics.Process/tests/ProcessCollectionTests.cs

```diff
@@ -12,7 +12,7 @@ public class ProcessCollectionTests : ProcessTestBase
         [Fact]
         public void TestModuleCollectionBehavior()
         {
-            ProcessModule[] mArray = _process.Modules.Cast<ProcessModule>().ToArray();
+            ProcessModule[] mArray = Process.GetCurrentProcess().Modules.Cast<ProcessModule>().ToArray();
```

Comment:
Not saying the fix is wrong, but regarding this:

> RemoteExecutorConsoleApp.exe, which is built for Any CPU, and Prefer 32bit property is not set on the project. So while running on 64 bit machines, it will start as 64 bit process

Won't the bitness of the process just be based on which runtime we are using? Our tests are all using a 32-bit runtime (at least for now). That's why the xUnit console process is 32-bit (it's actually compiled as AnyCPU 64-bit preferred). I would guess that the Remote Executor process is using the same runtime and therefore is probably also 32-bit.